src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryPartitionedUserTokenCacheAccessor.cs

```diff
@@ -0,0 +1,259 @@
+        {
+            string itemKey = item.GetKey().ToString();
+            string partitionKey = CacheKeyFactory.GetKeyFromCachedItem(item);
+
+            IdTokenCacheDictionary
+                .GetOrAdd(partitionKey, new ConcurrentDictionary<string, MsalIdTokenCacheItem>())[itemKey] = item;
+        }
+
+        public void SaveAccount(MsalAccountCacheItem item)
+        {
+            string itemKey = item.GetKey().ToString();
+            string partitionKey = CacheKeyFactory.GetKeyFromCachedItem(item);
+
+            AccountCacheDictionary
+                .GetOrAdd(partitionKey, new ConcurrentDictionary<string, MsalAccountCacheItem>())[itemKey] = item;
+        }
+
+        public void SaveAppMetadata(MsalAppMetadataCacheItem item)
+        {
+            string key = item.GetKey().ToString();
+            AppMetadataDictionary[key] = item;
+        }
+        #endregion
+
+        #region Get
+
+        public MsalIdTokenCacheItem GetIdToken(MsalAccessTokenCacheItem accessTokenCacheItem)
+        {
+            string partitionKey = CacheKeyFactory.GetIdTokenKeyFromCachedItem(accessTokenCacheItem);
+
+            IdTokenCacheDictionary.TryGetValue(partitionKey, out var partition);
+            if (partition != null && partition.TryGetValue(accessTokenCacheItem.GetIdTokenItemKey().ToString(), out var idToken))
+            {
+                return idToken;
+            }
+
+            _logger.WarningPii(
+                $"Could not find an id token for the access token with key {accessTokenCacheItem.GetKey()}",
+                $"Could not find an id token for the access token for realm {accessTokenCacheItem.TenantId} ");
+            return null;
+        }
+
+        public MsalAccountCacheItem GetAccount(MsalAccountCacheKey accountKey)
+        {
+            string partitionKey = CacheKeyFactory.GetKeyFromAccount(accountKey);
+
+            AccountCacheDictionary.TryGetValue(partitionKey, out var partition);
+            MsalAccountCacheItem cacheItem = null;
+            partition?.TryGetValue(accountKey.ToString(), out cacheItem);
+            return cacheItem;
+        }
+
+        public MsalAppMetadataCacheItem GetAppMetadata(MsalAppMetadataCacheKey appMetadataKey)
+        {
+            AppMetadataDictionary.TryGetValue(appMetadataKey.ToString(), out MsalAppMetadataCacheItem cacheItem);
+            return cacheItem;
+        }
+        #endregion
+
+        #region Delete
+        public void DeleteAccessToken(MsalAccessTokenCacheItem item)
+        {
+            string partitionKey = CacheKeyFactory.GetKeyFromCachedItem(item);
+
+            AccessTokenCacheDictionary.TryGetValue(partitionKey, out var partition);
+            if (partition == null || !partition.TryRemove(item.GetKey().ToString(), out _))
+            {
+                _logger.InfoPii(
+                    $"Cannot delete access token because it was not found in the cache. Key {item.GetKey()}.",
```

Comment:
I don't think this is Pii. Also what is the value in logging the cache key?

Reply:
We can't really ask customers for Pii logs, so let's try to not put any type of Pii logging where it isn't needed.

Reply:
I think home account id is PII